lib/elixir/lib/kernel/parallel_compiler.ex

```diff
@@ -110,17 +110,20 @@ defmodule Kernel.ParallelCompiler do
         :erlang.put(:elixir_compiler_file, file)
         :erlang.process_flag(:error_handler, Kernel.ErrorHandler)
 
-        exit(try do
-          _ = if output do
-            :elixir_compiler.file_to_path(file, output)
-          else
-            :elixir_compiler.file(file, Keyword.get(options, :dest))
+        result =
+          try do
+            _ = if output do
+              :elixir_compiler.file_to_path(file, output)
+            else
+              :elixir_compiler.file(file, Keyword.get(options, :dest))
+            end
+            :ok
+          catch
+            kind, reason ->
+              {kind, reason, System.stacktrace}
           end
-          {:shutdown, file}
-        catch
-          kind, reason ->
-            {:failure, kind, reason, System.stacktrace}
-        end)
+
+        send(parent, {:file_compiled, self(), file, result})
```

Comment:
This will cause an exit with reason `normal` on `{kind, reason, stack}`. If the file (spawn) linked any processes the links won't be broken by the exit signal.